packages/babylon/src/plugins/flow.js

```diff
@@ -2272,4 +2274,101 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         /* isAsync */ true,
       );
     }
+
+    readToken_mult_modulo(code: number): void {
+      // '%*'
+      let type = code === charCodes.asterisk ? tt.star : tt.modulo;
+      let width = 1;
+      let next = this.input.charCodeAt(this.state.pos + 1);
+      const exprAllowed = this.state.exprAllowed;
+
+      // Exponentiation operator **
+      if (code === charCodes.asterisk && next === charCodes.asterisk) {
+        width++;
+        next = this.input.charCodeAt(this.state.pos + 2);
+        type = tt.exponent;
+      }
+
+      if (next === charCodes.equalsTo && !exprAllowed) {
+        width++;
+        type = tt.assign;
+      }
+
+      if (
+        code === charCodes.asterisk &&
+        next === charCodes.slash &&
+        this.state.hasFlowComment
+      ) {
+        this.state.hasFlowComment = false;
+        this.state.pos += 2;
+        this.nextToken();
+        return;
+      }
+
+      this.finishOp(type, width);
+    }
+
+    skipBlockComment(): void {
```

Comment:
You can use `super.skipBlockComment()` to avoid duplicaitng its code.
e.g. probably
```js
skipBlockComment(): void {
  if (
    this.plugins.flow &&
    this.plugins.flowComments &&
    this.skipFlowComment()
   ) {
    this.hasFlowCommentCompletion();
    this.state.pos += this.skipFlowComment();
    this.state.hasFlowComment = true;
    return;
  }
  const startLoc = this.state.curPosition();
  const start = this.state.pos;

  let end;
  if (this.plugins.flow && this.state.hasFlowComment) {
    end = this.input.indexOf("*-/", (this.state.pos += 2));
    if (end === -1) this.raise(this.state.pos - 2, "Unterminated comment");
    this.state.pos = end + 3;
  }

  super.skipFlowComent();
}
```

Also for `readToken_mult_modulo`: if the surrent characters aren't `*/`, you can defer to `super.readToken_mult_modulo`.